src/tools/compiletest/src/main.rs

```diff
@@ -252,8 +252,24 @@ pub fn parse_config(args: Vec<String>) -> Config {
         }),
         logfile: matches.opt_str("logfile").map(|s| PathBuf::from(&s)),
         runtool: matches.opt_str("runtool"),
-        host_rustcflags: Some(matches.opt_strs("host-rustcflags").join(" ")),
-        target_rustcflags: Some(matches.opt_strs("target-rustcflags").join(" ")),
+        host_rustcflags: Some(
+            matches
+                .opt_strs("host-rustcflags")
+                .iter()
+                .map(|x| x.split_whitespace())
```

Comment:
To be clear, my thinking was that we wouldn't split on whitespace at all; callers would have to provide `--target-rustcflags` more than once if they want to specify multiple flags. I think all compiletest callers are programs today, so hopefully that's not too painful.

Reply:
This was required due to `x.py test` passing a singular `--target-rustcflags` call with multiple args separated by spaces. I'll find where that is coming from, and shift to the desired behavior of multiple `--target-rustcflags` calls